### PR TITLE
Rework GUI backend on Zig/MerJS and fix native regressions

### PR DESCRIPTION
**Pull request by pranavp311**
## Summary

This PR continues the GUI/backend migration work by replacing the old Tauri/Rust backend path with the Zig + MerJS native shell backend and fixes regressions found while dogfooding the native GUI.

> [!IMPORTANT]
> **Merge/release blocker: MerJS is not yet available as a normal dependency.**
>
> The GUI native shell currently depends on a local MerJS checkout via `gui/build.zig.zon`:
>
> ```zig
> .merjs = .{
>     .path = "../../../merjs",
> },
> ```
>
> That means this PR builds on the author machine because a sibling `merjs` checkout exists, but a fresh Codegraff checkout will not build the native GUI unless the reviewer also has the same local MerJS checkout.
>
> Before this PR can be considered merge/release-ready, we should do one of:
>
> 1. **Preferred:** merge/publish the corresponding MerJS PR, then update `gui/build.zig.zon` to a pinned `.url` + `.hash` dependency.
> 2. **Fallback:** vendor MerJS into this repo. Full vendoring is possible but noisy: the current MerJS repo has roughly **490 tracked files / 21MB**. A smaller minimal vendored subset may be possible but would need careful validation.
> 3. **Temporary alternative:** add MerJS as a git submodule, if maintainers are comfortable with that workflow.
>
> Until one of the above happens, treat this PR as dependent on the unmerged MerJS work.

### Backend + native shell rework

- Adds a MerJS-powered native GUI build/package path under `gui/build.zig` and `gui/native/main.zig`.
- Routes GUI API calls through a Zig backend runtime (`gui/src-backend/runtime.zig`) and MerJS route table (`gui/routes.zig`, `gui/api/stub.zig`).
- Removes the old `gui/src-tauri` backend from the GUI runtime path.
- Wires native shell actions through Mer where possible, including directory/file open flows, clipboard bridge commands, and app state persistence.
- Reconciles release updater UI as a native-safe stub until native desktop updates are wired; CLI updates remain available via `graff update`.
- Removes stale Tauri release-script references and updates GUI packaging flow around the native app bundle.

### GUI/session regressions fixed

- Restores chat sending and queued prompt behavior.
- Restores chat navigation and persisted chat/session loading.
- Adds session import/write support using harness-compatible session JSON files.
- Restores paste/file-drop handling in the prompt composer.
- Restores prompt-history attachment navigation, including Ultra prompt attachment-block parsing.
- Wires model catalog/provider settings into the Zig backend.
- Ranks model picker entries by usage and adds the UltraCode slash-choice picker.
- Shares Codex CLI auth / provider login detection with the GUI provider settings.
- Keeps draft/queued input usable while a request is running.
- Fixes segmented button border overlap/z-index behavior.
- Fixes sidebar footer/logo/settings overlap when many non-project chats are open and aligns the settings-view Codegraff logo hover highlight with the main sidebar logo.

### Terminal work

- Replaces the fake line-buffered terminal backend with a real PTY-backed shell.
- Vendors/adapts the working `openpty` pieces locally under `gui/src-backend/pty` so Codegraff does not depend on an external Zig package for PTY support.
- Adds Codegraff-owned PTY process lifecycle on top of `openpty`:
  - spawn shell in workspace cwd;
  - write raw input to PTY;
  - stream raw PTY output over existing SSE `terminal-output` events;
  - resize with `TIOCSWINSZ`;
  - terminate/close sessions cleanly.
- Fixes terminal sizing and removes output normalization that corrupted full-screen/TUI apps like `claude`/`codex`.
- Auto-copies terminal text selections with stale-guarded serialized clipboard writes and a dismissible toast.
- Preserves immediate terminal exit/error status instead of clearing it during startup replay.

### Harness invocation fixes

- Updates GUI backend to match the installed `graff --json` protocol.
- Removes unsupported `--resume <session>` CLI args from GUI-launched `graff` children.
- Uses the current legacy-compatible `set_model` request shape: `{ "type": "set_model", "name": "provider/model" }`.
- Keeps core harness changes aligned with the release baseline except for the intentional Kimi provider-login mapping/test update.

## Verification

Ran:

```bash
zig build test
cd gui && bunx tsc -b --pretty false
cd gui && bun test src/components/ui/ButtonGroup.test.ts src/components/promptHistoryNavigation.test.ts src/components/attachments/attachmentTypes.test.ts
cd gui && bunx eslint src/components/general-settings/UpdatesCard.tsx src/components/ui/ButtonGroup.tsx src/services/updater.ts src/components/workspace-board/hooks/useTerminalSession.ts src/components/ProjectSidebar.tsx
cd gui && bun run build
cd gui && zig build native
```

Manual / visual checks:

- GUI launches through MerJS native shell.
- Prompt submission reaches `graff --json` without the old `--resume` failure.
- PTY terminal opens real shell sessions.
- `claude`/`codex` style TUI output renders correctly after terminal sizing/raw output fixes.
- Sidebar footer no longer overlays the Projects header/action when the chats list is long.
- Settings-view Codegraff logo uses the same hover/highlight treatment as the main sidebar logo.
- Kuri/Chrome was used for a browser visual smoke check of the rebuilt GUI bundle where possible.

### CODE REVIEW

**Comment by pranavp311** [comment]
Updated this PR with Codegraff GUI fixes in b269fad (pushed to backend-zig-merjs-regression-fixes):

- Stabilized Enter submit flow by making Enter use the textarea DOM value, pass explicit workspace/conversation context, and guard duplicate rapid submits.
- Added optimistic user-message rendering with rollback on send failure.
- Fixed chat auto-titling/persistence so CLI-compatible session files import/retitle from the first user message and persist durable titles.
- Hid empty New Chat drafts from the Chats sidebar and stopped writing empty draft session files before the first message.
- Added Codegraff-side Cmd+W wiring via the native bridge client/manifest.

Validated in Codegraff GUI:
- bunx tsc -b
- zig build

Note: native shell bridge implementation for window.close is committed separately to the MerJS PR (#100), not this Codegraff PR.

**Comment by justrach** [comment]
@codegraff-bot You're checked out on this PR's branch. Run `git diff origin/main...HEAD --stat` and inspect the key changed files, then give a concise, actionable code review: correctness risks, design concerns, and the most important things to fix before merge. Reference specific files.

**Comment by justrach** [comment]
@codegraff-bot please review this PR. Focus on: the MerJS local-path dependency in gui/build.zig.zon (fresh-checkout buildability), the Zig backend runtime + MerJS route table, the new PTY-backed terminal lifecycle, and any native GUI regressions in the reworked shell.

**Comment by justrach** [comment]
@codegraff-bot fix the C2 shutdown use-after-free: detached terminal reader threads (terminalReaderMain) and graff turn threads (graffTurnThread) call thread.detach() and then touch rt internals (rt.mutex, rt.terminals, rt.emitSseEvent, rt.allocator) after Shell.run() returns and Runtime.deinit() frees them. Store the std.Thread handles in TerminalSessionState and the graff session state, set closing=true + SIGHUP before deinit, and join every spawned thread during cleanup instead of detaching, so no thread outlives the Runtime. Scope this to the shutdown/thread-lifecycle path only.
